File: 167. Two Sum II, Sorted Array/BinarySearchSolution.java
class BinarySearchSolution 
{

    public int binarySearch(int[] nums, int target, int left, int right)
    {
        int mid = (left+right)/2;

		//Base condition
        if(left > right)
        {
            return -10000;
        }

        if(nums[mid] == target)
        {
            return mid;
        }

        else if(nums[mid] > target)
        {
            return binarySearch(nums,target,left, mid-1);
        }

        else
        {
            return binarySearch(nums,target,mid+1, right);
        }
    }

    public int[] twoSum(int[] numbers, int target) 
    {
        
        for (int i = 0; i < numbers.length ;  i++)
        {
            int remaining = target - numbers[i];

            int index = binarySearch(numbers,remaining,i+1, numbers.length -1);

            if (index != -10000)
            {
                return new int[] {i+1,index+1};
            }
            
        }
		//To avoid compilation error
        return new int[2];
    }
}

File: 167. Two Sum II, Sorted Array/BinarySearchSolution.py
class BinarySearchSolution:

    #Binary Search function
    def binarySearch(self, left, right, nums, target):
        
        #Base condition in case the index is not found in this iteration
        if left > right:
            return -1

        mid = (left+right)//2

        if nums[mid] == target:
            return mid

        elif nums[mid] > target:
            return self.binarySearch(left, mid-1, nums, target)

        else:
            return self.binarySearch(mid+1, right, nums, target)


    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        length = len(numbers)
	
        #Iterating over the list using the index as we have to return indices here
        for i in range(len(numbers)):

            #Calculating the result of the target-nums[i] to find in the list
            remaining = target - numbers[i]

            #Using Binary search to get the index of the array
            j = self.binarySearch(i+1, length-1, numbers, remaining)

            #If index is not negative, return the indices+1 as this is a 1 based index array.
            if j != -1:
                return [i+1,j+1]
